pkg/apiserver/registry/security/workloadidentity/storage/jwt.go

```diff
@@ -0,0 +1,291 @@
+// SPDX-FileCopyrightText: 2024 SAP SE or an SAP affiliate company and Gardener contributors
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package storage
+
+import (
+	"crypto"
+	"crypto/ecdsa"
+	"crypto/elliptic"
+	"crypto/rsa"
+	"crypto/sha256"
+	"crypto/x509"
+	"encoding/base64"
+	"fmt"
+	"time"
+
+	"github.com/go-jose/go-jose/v4"
+	"github.com/go-jose/go-jose/v4/jwt"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+
+	gardencorev1beta1 "github.com/gardener/gardener/pkg/apis/core/v1beta1"
+	securityapi "github.com/gardener/gardener/pkg/apis/security"
+	gardenerutils "github.com/gardener/gardener/pkg/utils/gardener"
+)
+
+const (
+	keyUsageSig string = "sig"
+)
+
+type gardenerClaims struct {
+	Gardener gardener `json:"gardener.cloud,omitempty"`
+}
+
+type gardener struct {
+	WorkloadIdentity ref     `json:"workloadIdentity,omitempty"`
+	Shoot            *ref    `json:"shoot,omitempty"`
+	Project          *ref    `json:"project,omitempty"`
+	Seed             *ref    `json:"seed,omitempty"`
+	Garden           *garden `json:"garden,omitempty"`
+}
+
+type ref struct {
+	Name      string `json:"name,omitempty"`
+	Namespace string `json:"namespace,omitempty"`
+	UID       string `json:"uid,omitempty"`
+}
+
+type garden struct {
+	ID string `json:"id,omitempty"`
+}
+
+// issueToken generates the JSON Web Token based on the provided configurations.
+func (r *TokenRequestREST) issueToken(tokenRequest *securityapi.TokenRequest, workloadIdentity *securityapi.WorkloadIdentity) (string, *time.Time, error) {
+	signer, err := getSigner(r.signingKey)
```

Comment:
I think that this signer can be initialised once and reused across requests.

Reply:
Yes, will do it.